cmd/kubeadm/app/phases/kubeconfig/kubeconfig_test.go

```diff
@@ -675,7 +677,138 @@ func TestValidateKubeconfigsForExternalCA(t *testing.T) {
+			CertificatesDir: pkiDir,
+		},
+		LocalAPIEndpoint: kubeadmapi.APIEndpoint{
+			BindPort:         1234,
+			AdvertiseAddress: "1.2.3.4",
+		},
+	}
+
+	// creates CA, write to pkiDir and remove ca.key to get into external CA condition
+	caCert, caKey := certstestutil.SetupCertificateAuthority(t)
+
+	// create a config with a CA cert containing multiple certificates
+	intermediateCACert1, intermediateCAKey1 := certstestutil.SetupIntermediateCertificateAuthority(t, caCert, caKey)
+	intermediateCACert2, intermediateCAKey2 := certstestutil.SetupIntermediateCertificateAuthority(t, intermediateCACert1, intermediateCAKey1)
+
+	var caCertBundle []*x509.Certificate
+	caCertBundle = append(caCertBundle, caCert, intermediateCACert1, intermediateCACert2)
+	caCertBytes, _ := pkiutil.EncodeCertBundlePEM(caCertBundle)
+
+	clusterName := "myOrg1"
+
+	// Create a kube config and assign the CA data to the CA bundle, issued from root CA
+	multipleCAConfigRootCAIssuer := setupdKubeConfigWithClientAuth(t, caCert, caKey, "https://1.2.3.4:1234", "test-cluster", clusterName)
+	multipleCAConfigRootCAIssuer.Clusters[clusterName].CertificateAuthorityData = caCertBytes
+
+	// Create a kube config and assign the CA data to the CA bundle, issued from intermediate CA 1
+	multipleCAConfigIntermediateCA1Issuer := setupdKubeConfigWithClientAuth(t, intermediateCACert1, intermediateCAKey1, "https://1.2.3.4:1234", "test-cluster", clusterName)
+	multipleCAConfigIntermediateCA1Issuer.Clusters[clusterName].CertificateAuthorityData = caCertBytes
+
+	// Create a kube config and assign the CA data to the CA bundle, issued from intermediate CA 2
+	multipleCAConfigIntermediateCA2Issuer := setupdKubeConfigWithClientAuth(t, intermediateCACert2, intermediateCAKey2, "https://1.2.3.4:1234", "test-cluster", clusterName)
+	multipleCAConfigIntermediateCA2Issuer.Clusters[clusterName].CertificateAuthorityData = caCertBytes
+
+	// create a config with a CA cert containing multiple certificates, omitting the root CA
+	var caCertBundleNoRootCA []*x509.Certificate
+	caCertBundleNoRootCA = append(caCertBundleNoRootCA, intermediateCACert1, intermediateCACert2)
+	caCertBytesNoRootCA, _ := pkiutil.EncodeCertBundlePEM(caCertBundleNoRootCA)
+
+	// Create a kube config and assign the CA data to the CA bundle, issued from intermediate CA 2,
+	// CA is missing root CA certificate.
+	multipleCAConfigNoRootCA := setupdKubeConfigWithClientAuth(t, intermediateCACert2, intermediateCAKey2, "https://1.2.3.4:1234", "test-cluster", clusterName)
+	multipleCAConfigNoRootCA.Clusters[clusterName].CertificateAuthorityData = caCertBytesNoRootCA
+
+	if err := pkiutil.WriteCertBundle(pkiDir, kubeadmconstants.CACertAndKeyBaseName, caCertBundleNoRootCA); err != nil {
+		t.Fatalf("failure while saving CA certificate: %v", err)
+	}
+
+	tests := map[string]struct {
+		filesToWrite  map[string]*clientcmdapi.Config
+		initConfig    *kubeadmapi.InitConfiguration
+		expectedError bool
```

Comment:
all expected errors in this test are "false", i guess we should have at least one with "true"?

Reply:
Added some negative tests. Reworked some additional logic. Will explain in more detail in a comment.